### PR TITLE
add pixi export

### PR DESCRIPTION
**Pull request by PSeitz**
#191

---
This change is [Reviewable](https://reviewable.io/reviews/juliandescottes/piskel/581)

### CODE REVIEW

**Comment by PSeitz** [comment]
This creates a zip with the spritesheet and a json metadatafile in the pixijs format.
But to download only the json metadatafile and separate it from the png spritesheet download may be better.

screen1

**Comment by juliandescottes** [comment]
Thanks @PSeitz this looks great! 

I was about to suggest moving this to the "other" tab, but I guess it can be useful to set the number of columns for the export here, so let's leave it in the "PNG" tab. 

I think we should open another issue to rework a bit the UI of this tab. Maybe group the image download / data-uri options (or even remove the data-uri button completely ... this one got released by mistake, I wanted to have it only for development initially).

I am not really using Pixi personally, so if @hassanshaikley wants to also give some feedback that'd be great!

**Comment by juliandescottes** [comment]
Code change looks good. Will merge soon.

File: src/js/controller/settings/exportimage/PngExportController.js [hunk 1]
```diff
@@ -156,6 +158,52 @@
...
+        'spriteSourceSize': {'x': 0,'y': 0,'w': width,'h': height},
+        'sourceSize': {'w': width,'h': height}
+      };
+      frames[name + i + '.png'] = frame;
+    }
+
+    var json = {
+      'frames': frames,
+      'meta': {
+        'app': 'http://www.pisker.com',
```
Comment (juliandescottes) [inline src/js/controller/settings/exportimage/PngExportController.js]: was this supposed to be piskelapp.com ? 

I guess we can just link to https://github.com/juliandescottes/piskel/

File: src/js/controller/settings/exportimage/PngExportController.js [hunk 2]
```diff
@@ -156,6 +158,52 @@
...
+      var column = i % this.getColumns_();
+      var row = (i - column) / this.getColumns_();
+      var frame = {
+        'frame': {'x': width * column,'y': height * row,'w': width,'h': height},
+        'rotated': false,
+        'trimmed': false,
+        'spriteSourceSize': {'x': 0,'y': 0,'w': width,'h': height},
+        'sourceSize': {'w': width,'h': height}
+      };
+      frames[name + i + '.png'] = frame;
```
Comment (juliandescottes) [inline src/js/controller/settings/exportimage/PngExportController.js]: I quickly tried this modifying the examples at https://github.com/pixijs/examples

The code was expecting frame numbers to be padded (ie piskel_name1.png -> piskel_name01.png). I'm not sure if this is something specific to the examples or that is more broadly expected for pixijs spritesheets?

**Comment by hassanshaikley** [comment]
This is awesome! : )

**Comment by juliandescottes** [comment]
Thanks for the feedback @hassanshaikley 

Merging and opening follow up issues to track my review comments.
Thanks for the great work @PSeitz !

File: src/js/controller/settings/exportimage/PngExportController.js (hunk 2 above)
Comment (PSeitz) [inline src/js/controller/settings/exportimage/PngExportController.js]: I don't know why they do this, in my example it works fine and the code is simpler. When the json+spritesheet is loaded, you usually reference the images by their names in pixijs.

File: src/js/controller/settings/exportimage/PngExportController.js (hunk 1 above)
Comment (PSeitz) [inline src/js/controller/settings/exportimage/PngExportController.js]: Yes, that should be fine

File: src/js/controller/settings/exportimage/PngExportController.js (hunk 2 above)
Comment (juliandescottes) [inline src/js/controller/settings/exportimage/PngExportController.js]: Yes, looks like this was really specific to this example. I think your implementation is fine.

**Pull request merged**